Sources/Sentry/SentryTracer.m

```diff
@@ -631,19 +631,31 @@ - (void)finishInternal
 - (void)captureTransactionWithProfile:(SentryTransaction *)transaction
                        startTimestamp:(NSDate *)startTimestamp
 {
-    SentryEnvelopeItem *profileEnvelopeItem
-        = sentry_traceProfileEnvelopeItem(transaction, startTimestamp);
+    // This code can run on the main thread, and the profile serialization can take a couple of
+    // milliseconds. Therefore, we move this to a background thread to avoid potentially blocking
+    // the main thread.
+    __weak SentryTracer *weakSelf = self;
```

Comment:
h: I don’t think this is desirable, as it makes it possible (and actually easy) to lose the profiling. I believe it’s okay to hold a reference to self until the capture finishes, or at least hold a reference to the hub, which is the only thing we need to use inside the block.

Reply:
That makes sense. Good catch thanks 👍